### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by Shahar-Y**

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+      // If value found, update the cache in background and return the cached value.
+      retrieveFunction()
+        .then(async (result: string) => {
+          logger.log('Cached value found. Updating cache');
+          await this.safeSetCache(key, result, expire);
+        })
+        .catch((err: Error) => {
+          criticalLog(err);
+          callbackFunction();
+          throw err;
```
Comment (galGeva123) [inline src/index.ts:102]: remove the err, you can put it in the callbackFunction

File: src/index.ts [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+      if (!cachedValue) {
+        const result = await retrieveFunction();
+        await this.safeSetCache(key, result, expire);
+        return result;
+      }
+
+      // If value found, update the cache in background and return the cached value.
+      retrieveFunction()
+        .then(async (result: string) => {
+          logger.log('Cached value found. Updating cache');
```
Comment (galGeva123) [inline src/index.ts]: remove

File: src/example/example.ts [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+import { TryCache } from '../index';
+
+(async () => {
+  const cb = new TryCache(defaults.redis, { silent: false, expire: 20 });
+  cb.initTryCache();
+
+  const myKey = 'myKey';
+
+  console.log('Retrieving data for the first time');
+  const startNoCache = new Date();
```
Comment (galGeva123) [inline src/example/example.ts]: use - Console.time(label)
Console.endtime(label)

File: src/example/example.ts [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+function sleep(ms: number): Promise<void> {
+  return new Promise((resolve) => setTimeout(resolve, ms));
+}
+
+// todo: read
+process.on('SIGINT', async () => {
+  console.log('Closing gracefully');
+  await Redis.quit();
+  console.log('Shutdown complete');
+  process.exit(0);
```
Comment (galGeva123) [inline src/example/example.ts:74]: when i run this, didnt close the program for me

Comment (Shahar-Y) [reply src/example/example.ts:74]: It closes gracefully when you press ctrl+C

**Pull request merged**